server/src/test/java/org/elasticsearch/cluster/health/ClusterIndexHealthTests.java

```diff
@@ -141,187 +141,47 @@ protected Predicate<String> getRandomFieldsExcludeFilter() {
-                    instance.getUnassignedShards(),
-                    instance.getUnassignedPrimaryShards(),
-                    instance.getActivePrimaryShards() + between(1, 10),
-                    instance.getStatus(),
-                    instance.getShards()
-                );
-            case "status":
-                ClusterHealthStatus status = randomFrom(
-                    Arrays.stream(ClusterHealthStatus.values()).filter(value -> value.equals(instance.getStatus()) == false).toList()
-                );
-                return new ClusterIndexHealth(
-                    instance.getIndex(),
-                    instance.getNumberOfShards(),
-                    instance.getNumberOfReplicas(),
-                    instance.getActiveShards(),
-                    instance.getRelocatingShards(),
-                    instance.getInitializingShards(),
-                    instance.getUnassignedShards(),
-                    instance.getUnassignedPrimaryShards(),
-                    instance.getActivePrimaryShards(),
-                    status,
-                    instance.getShards()
-                );
-            case "shards":
-                Map<Integer, ClusterShardHealth> map;
-                if (instance.getShards().isEmpty()) {
-                    map = Collections.singletonMap(0, ClusterShardHealthTests.randomShardHealth(0));
-                } else {
-                    map = new HashMap<>(instance.getShards());
-                    map.remove(map.keySet().iterator().next());
-                }
-                return new ClusterIndexHealth(
-                    instance.getIndex(),
-                    instance.getNumberOfShards(),
-                    instance.getNumberOfReplicas(),
-                    instance.getActiveShards(),
-                    instance.getRelocatingShards(),
-                    instance.getInitializingShards(),
-                    instance.getUnassignedShards(),
-                    instance.getUnassignedPrimaryShards(),
-                    instance.getActivePrimaryShards(),
-                    instance.getStatus(),
-                    map
-                );
-            default:
-                throw new UnsupportedOperationException();
+        String index = instance.getIndex();
+        int numberOfShards = instance.getNumberOfShards();
+        int numberOfReplicas = instance.getNumberOfReplicas();
+        int activeShards = instance.getActiveShards();
+        int relocatingShards = instance.getRelocatingShards();
+        int initializingShards = instance.getInitializingShards();
+        int unassignedShards = instance.getUnassignedShards();
+        int unassignedPrimaryShards = instance.getUnassignedPrimaryShards();
+        int activePrimaryShards = instance.getActivePrimaryShards();
+        ClusterHealthStatus status = instance.getStatus();
+        Map<Integer, ClusterShardHealth> shards = instance.getShards();
```

Comment:
I think you're missing a switch branch for the shards.

Reply:
You're right and interesting that the tests didn't pick up on that :-/ I ran them several times as did buildkite...

Anyway have added that extra switch in.

Reply:
Tip: you can run tests with [`-Dtests.iters=1000`](https://github.com/elastic/elasticsearch/blob/main/TESTING.asciidoc#repeats-all-tests-of-classname-n-times) to run them as often as you want. Alternatively, you can also repeat tests with `@com.carrotsearch.randomizedtesting.annotations.Repeat(iterations=1000)`.